Translate "Goodbye, clean code" to spanish (#577)
gaearon/overreacted.io: src/pages/goodbye-clean-code/index.es.md

```

¡El código ocupaba la mitad y la duplicación se había ido completamente! Entonces estaba *limpio*. Si queríamos cambiar el comportamiento para una dirección en particular, podíamos hacerlo en un solo lugar, en vez de actualizar todos los métodos que estaban duplicados.

Ya era tarde en la noche (me dejé llevar). Subí mis cambios de refactorización a master y me fuí a la cama, orgulloso de como pude desenredar el código desordenado de mi colega.

## La mañana siguiente

... no fue como esperaba.

Mi jefe me invitó a una reunión uno-a-uno donde cortésmente me pidieron que revierta mis cambios. Estaba atónito. ¡El código anterior era un desastre, y el mio era *limpio*!.

Cumplí a regañadientes, pero me tomó años darme cuenta que tenían razón.

## Es una fase

Obsesionarse con el "código limpio" y eliminar toda duplicación es una fase por la que muchos de nosotros pasamos. Cuando no nos sentimos seguros de nuestro código, es tentador vincular nuestro sentido de autoestima y orgullo profesional a algo que se puede medir. Un conjunto de reglas de un linter,un esquema de nombres, una estructura de archivos, una falta de duplicación.

> No puedes automatizar la eliminación de código repetido, pero se hace más facil con la práctica. Usualmente puedes darte cuenta que hay más o menos de esto después de cada cambio. Como resultado, eliminar duplicación se siemte como mejorar alguna metrica objetiva del código. Peor, esto desordena con el sentido de identidad de las personas: *Soy el tipo de persona que escribe código limpio*. Es tan poderoso como cuanquier otro tipo de auto-engaño.

Reviewer: ```suggestion
No puedes automatizar la eliminación de código repetido, pero se hace más fácil con la práctica. Usualmente puedes darte cuenta que hay más o menos de esto después de cada cambio. Como resultado, eliminar duplicación se siemte como mejorar alguna métrica objetiva del código. Peor, esto desordena con el sentido de identidad de las personas: *Soy el tipo de persona que escribe código limpio*. Es tan poderoso como cualquier otro tipo de autoengaño.
```

> Una vez que aprendemos como crear [abstracciones](https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction), es tentador dejarse llevar por esa habilidad y sacar abstracciones de la nada en cualquier momento que vemos código repetido. Despues de muchos años codificando, vemos repetición *en todo lado* -- y abstraer es nuestro nuevo super poder. Si alguien nos dice que la abstracción es una *virtud*, le creemos. Y empezamos a juzgar a otros por no redir culto a la "limpieza en el código".

Reviewer: ```suggestion
Una vez que aprendemos como crear [abstracciones](https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction), es tentador dejarse llevar por esa habilidad y sacar abstracciones de la nada en cualquier momento que vemos código repetido. Despues de muchos años codificando, vemos repetición *en todos lados* -- y abstraer es nuestro nuevo superpoder. Si alguien nos dice que la abstracción es una *virtud*, le creemos. Y empezamos a juzgar a otros por no rendir culto a la "limpieza en el código".
```

Ahora veo como mi "refactorización" era un desastre por dos razones:

> * Primero, no hablé con la persona que lo escribió. Reescribí el código y lo subí sin su aporte. Aunque era una mejora (cosa que ya no considero asi), es una manera terrible de hacerlo. Un equipo de ingeniería sano está constantemente *construyendo confianza*. Reescribir el código de tu compañero de equipo sin una discusión previa es un gran golpe para su habilidad como equipo de colaborar efectivamente en una base de código.

Reviewer: ```suggestion
* Primero, no hablé con la persona que lo escribió. Reescribí el código y lo subí sin su aporte. Aunque era una mejora (cosa que ya no considero así), es una manera terrible de hacerlo. Un equipo de ingeniería sano está constantemente *construyendo confianza*. Reescribir el código de tu compañero de equipo sin una discusión previa es un gran golpe para su habilidad como equipo de colaborar efectivamente en una base de código.
```

* Segundo, nada es gratis. Mi código intercambiaba la habilidad de cambiar los requerimientos por menos duplicidad de código, y ese no era un buen intercambio. Por ejemplo, luego necesitamos muchos casos y comportamientos especiales para los diferentes controles en las diferentes figuras. Mi abstracción se hubiera convertido mucho más desordenado para lograr eso a la larga, mientras que la versión original "desordenada" pudo afrontar esos cambios de manera fácil.

¿Estoy diciendo que debemos escribir código "sucio"? No. Estoy sugiriendo que pensemos profundamente a que nos referimos cuando decimos "limpio" o "sucio". ¿Te da un sentimiento de revolución? ¿Correctitud? ¿Belleza? ¿Elegancia? ¿Qué tan seguro estás de que puedes darles un nombre a los resultados de ingeniería concretos correspondientes a esas cualidades? ¿Cómo afectan exactamente a la manera en la que el código está escrito y es [modificado](/optimized-for-change/)?

> Estoy seguro que no pensaste profundamente acerca de estas cosas. Yo pensaba mucho acerca de como *lucía* el código -- pero no acerca de como *evolucionaba* con un equipo de humanos débiles.

Reviewer: Hola @Veuge, releyendo el artículo me resulta un poco extraña esta traducción de «*squishy humans*». Aunque «humanos débiles» es una traducción literal del término me parece que puede tener una connotación un poco negativa en español. ¿Quizá algo como «simples humamos» podría funcionar? Aunque en este caso creo que pierde un poco del matiz de la frase original. ¡Si tienes una mejor opción no dudes en darla! Nota que esto no bloquea el PR, podríamos dejarlo como está, solamente quisiera consultar tu opinión antes de mezclarlo.

Author: Hola! estuve buscando traducciones para _squishy humans,_ no encontré mas que adjetivos como blando, debil, suave. Ninguno de esos me parece aplicable al contexto. Estoy de acuerdo con _simples humanos,_ refleja mejor a lo que Dan se refiere en ese parrafo.

Reviewer: !Muchas gracias por la respuesta! Mezclamos entonces :rocket:

> Codificar es un viaje. Piensa que tan lejos has llegado desde tu primera linea de código hasta donde estas ahora. Reconozco que fue una alegría ver por primera vez como extraer una función o refactorizar una clase puede hacer el código mas simple. Si te enorgulleces de tu oficio, es tentador perseguir la limpieza de código. Hazlo por un tiempo.

Reviewer: ```suggestion
Codificar es un viaje. Piensa que tan lejos has llegado desde tu primera linea de código hasta donde estás ahora. Reconozco que fue una alegría ver por primera vez como extraer una función o refactorizar una clase puede hacer el código mas simple. Si te enorgulleces de tu oficio, es tentador perseguir la limpieza de código. Hazlo por un tiempo.
```

Pero no te quedes ahi. No seas un fanático del código limpio. El código limpio no es una meta. Es un intento de dar sentido a la inmensa complejidad de sistemas con los que lidiamos. Es un mecanismo de defensa cuando no estas seguro de como un cambio puede afectar una base de código pero necesitas guía en un mar de incognitas.

> Deja que el código limpio te guíe. *Luego dejalo ir.*

Reviewer: ```suggestion
Deja que el código limpio te guíe. *Luego déjalo ir.*
```
